### PR TITLE
Change the default sort of the meetings #41

### PR DESCRIPTION
**Pull request by ignatescustefan**
*the meeting on the first page are displayed in asc order, filter by the current date and the current date plus one year.
*also, the user can see all the meetings created.

### CODE REVIEW

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 1]
```diff
@@ -51,20 +51,39 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
...
-  'rowCount': 15,
-  'maxPages': 10,
-  'selectedColumn': 'startDate',
-  'defaultOrder': 'desc'
-})
-#set ($columns = ['doc.title', 'leader', 'place', 'startDate', 'durationDisplay', 'status', '_actions'])
+#set ($currentDate = $xwiki.jodatime.dateTime.millis)
+#set ($endDate = $xwiki.jodatime.dateTime.plusYears(1).millis)
+#set ($parameter= $request.getParameter("meeting"))
+#if($parameter=="all")
```
Comment (arcilli) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:62]: Wrong indentation: `#if ($parameter == "all")`

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 2]
```diff
@@ -51,20 +51,39 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
...
-  'tagCloud': true,
-  'rowCount': 15,
-  'maxPages': 10,
-  'selectedColumn': 'startDate',
-  'defaultOrder': 'desc'
-})
-#set ($columns = ['doc.title', 'leader', 'place', 'startDate', 'durationDisplay', 'status', '_actions'])
+#set ($currentDate = $xwiki.jodatime.dateTime.millis)
+#set ($endDate = $xwiki.jodatime.dateTime.plusYears(1).millis)
+#set ($parameter= $request.getParameter("meeting"))
```
Comment (arcilli) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:61]: A space before `=` is missing.

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 3]
```diff
@@ -51,20 +51,39 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
...
+    'className': 'Meeting.Code.MeetingClass',
+    'translationPrefix': 'meeting.livetable.',
+    'tagCloud': true,
+    'rowCount': 15,
+    'maxPages': 10,
+    'selectedColumn': 'startDate',
+    'defaultOrder': 'desc'
+  })
+#else
+  $services.localization.render('contrib.meeting.home.initialMessage',['[[**','&gt;&gt;||queryString="meeting=all"]]'])
```
Comment (oanalavinia) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:74]: Code style

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 4]
```diff
@@ -51,20 +51,39 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
...
+  #set ($options = {
+    'className': 'Meeting.Code.MeetingClass',
+    'translationPrefix': 'meeting.livetable.',
+    'tagCloud': true,
+    'rowCount': 15,
+    'maxPages': 10,
+    'selectedColumn': 'startDate',
+    'extraParams':"&amp;endDate=${currentDate}-${endDate}",
```
Comment (oanalavinia) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:82]: A space would be needed after `:`, like you did for the others

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 5]
```diff
@@ -51,20 +51,39 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
...
+    'defaultOrder': 'asc'
+  })
+#end
+#set ($columns = ['doc.title', 'leader', 'place', 'startDate','endDate', 'durationDisplay', 'status', '_actions'])
```
Comment (oanalavinia) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:86]: You missed a space between startDate and endDate

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml (hunk 1 above)
Comment (oanalavinia) [reply ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:62]: Single quotes should also be used `'all'`, same as for `'meeting'`

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml (hunk 4 above)
Comment (oanalavinia) [reply ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:82]: I see that for both declarations of `$options` there are some properties that remain the same, and only a few change depending on `$parameter`. Maybe you could change only some properties inside your `if` to not duplicate code

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml (hunk 2 above)
Comment (oanalavinia) [reply ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:61]: You only use `$parameter` once, so maybe you don't need a variable

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 6]
```diff
@@ -32,7 +32,7 @@
   <author>xwiki:XWiki.Admin</author>
   <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
   <date>1547210090000</date>
-  <contentUpdateDate>1547210090000</contentUpdateDate>
+  <contentUpdateDate>1547210090000</contentUpdateDate
```
Comment (oanalavinia) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:35]: You removed the end tag by mistake

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 7]
```diff
@@ -60,11 +61,20 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
   'translationPrefix': 'meeting.livetable.',
   'tagCloud': true,
   'rowCount': 15,
-  'maxPages': 10,
-  'selectedColumn': 'startDate',
-  'defaultOrder': 'desc'
+  'maxPages': 10
 })
-#set ($columns = ['doc.title', 'leader', 'place', 'startDate', 'durationDisplay', 'status', '_actions'])
+#if ($request.getParameter("meeting") == 'all')
```
Comment (oanalavinia) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:66]: I think that you should use single quotes for `'meeting'` as well, like you did for `'all'`

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 8]
```diff
@@ -60,11 +61,20 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
   'translationPrefix': 'meeting.livetable.',
   'tagCloud': true,
   'rowCount': 15,
-  'maxPages': 10,
-  'selectedColumn': 'startDate',
-  'defaultOrder': 'desc'
+  'maxPages': 10
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:64]: In the https://extensions.xwiki.org/xwiki/bin/view/Extension/Livetable+Macro#HParameter24options, you will find that the default for `maxPages` is `10`, so you can remove it from here, as it is redundant.

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 9]
```diff
@@ -51,6 +51,7 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
   'leader': {"type":"text","size":10,"html":true},
   'place': {"type":"text","size":10},
   'startDate': {"type":"text","size":10},
+  'endDate': {"type":"hidden"},
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:54]: As an improvement, separate from this PR, you can update the livetable definition to use the recommended code style: replacing double quotes `"` with single quotes `'` and adding a blank space after colon `:`. So you will have for this line
`'endDate': {'type': 'hidden'},`

File: ui/src/main/resources/Meeting/Code/WebHomeSheet.xml [hunk 10]
```diff
@@ -60,11 +61,20 @@ $services.localization.render('contrib.meeting.home.presentation', ['[[**', '**&
...
-  'defaultOrder': 'desc'
+  'maxPages': 10
 })
-#set ($columns = ['doc.title', 'leader', 'place', 'startDate', 'durationDisplay', 'status', '_actions'])
+#if ($request.getParameter('meeting') == 'all')
+  $services.localization.render('contrib.meeting.home.allMeetingsMessage', ['[[', "&gt;&gt;${doc}]]"])
+  #set ($options.selectedColumn = 'startDate')
+  #set ($options.defaultOrder = 'desc')
+#else
+  $services.localization.render('contrib.meeting.home.initialMessage', ['[[**', '&gt;&gt;||queryString="meeting=all"]]'])
```
Comment (acotiuga) [inline ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:71]:
```suggestion
  $services.localization.render('contrib.meeting.home.initialMessage', ['[[**', "&gt;&gt;||queryString='meeting=all']]"])
```

Comment (acotiuga) [reply ui/src/main/resources/Meeting/Code/WebHomeSheet.xml:71]: I would use the `"` as wrapper of `'` instead.

**Comment by ignatescustefan** [comment]
Here are the screenshots from the application after implementing the changes:
Meetings
Meeting_all

**Pull request merged**